src/m365/commands/setup.ts

```diff
@@ -106,59 +205,218 @@ class SetupCommand extends AnonymousCommand {
+    let clientCertificatePassword: string | undefined;
 
-      if (!this.verbose) {
-        await logger.logToStderr('');
-        await logger.logToStderr(chalk.green('DONE'));
+    const clientId = await cli.promptForInput({
+      message: 'Client ID:',
+      /* c8 ignore next */
+      validate: value => value.length > 0
+    });
+    const tenantId = await cli.promptForInput({
+      message: 'Tenant ID (leave common if the app is multitenant):',
+      default: 'common'
+    });
+    const clientSecret = await cli.promptForInput({
+      message: 'Client secret (leave empty if you use a certificate or a public client):'
+    });
+    if (!clientSecret) {
+      clientCertificateFile = await cli.promptForInput({
+        message: `Path to the client certificate file (leave empty if you want to specify a base64-encoded certificate string):`
+      });
+      if (!clientCertificateFile) {
+        clientCertificateBase64Encoded = await cli.promptForInput({
+          message: 'Base64-encoded certificate string:'
+        });
       }
+      if (clientCertificateFile || clientCertificateBase64Encoded) {
+        clientCertificatePassword = await cli.promptForInput({
+          message: 'Password for the client certificate (leave empty if the certificate is not password-protected):'
+        });
+      }
+    }
+
+    return {
+      clientId,
+      tenantId,
+      clientSecret,
+      clientCertificateFile,
+      clientCertificateBase64Encoded,
+      clientCertificatePassword
+    };
+  }
+
+  private async createNewEntraApp(preferences: Preferences, logger: Logger): Promise<AppInfo> {
+    if (!await cli.promptForConfirmation({
+      message: 'CLI for Microsoft 365 will now sign in to your Microsoft 365 tenant as Microsoft Azure CLI to create a new app registration. Continue?',
+      default: false
+    })) {
+      throw new Error();
     }
+
+    // logout, just in case
+    await auth.clearConnectionInfo();
+    // setup auth
+    auth.connection.authType = AuthType.Browser;
+    // Microsoft Azure CLI app ID
+    auth.connection.appId = '04b07795-8ddb-461a-bbee-02f9e1bf7b46';
+    auth.connection.tenant = 'common';
+    await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
+    auth.connection.active = true;
+
+    const options: AppCreationOptions = {
+      allowPublicClientFlows: true,
+      apisDelegated: (preferences.newEntraAppScopes === NewEntraAppScopes.All ? allScopes : minimalScopes).join(','),
+      implicitFlow: false,
+      multitenant: false,
+      name: 'CLI for Microsoft 365',
+      platform: 'publicClient',
+      redirectUris: 'http://localhost,https://localhost,https://login.microsoftonline.com/common/oauth2/nativeclient'
+    };
+    const apis = await entraApp.resolveApis({
+      options,
+      logger,
+      verbose: this.verbose,
+      debug: this.debug
+    });
+    const appInfo: AppInfo = await entraApp.createAppRegistration({
```

Comment:
Good point @Adam-it, it could help if we would be able to specify or override the name. Especially when working with customer tenants this could be useful...

Reply:
What's the scenario for creating a new app multiple times? Each app has the same configuration and is single tenant, so there's really no need to have multiple IMHO. If you need more control, you can always choose `entra app add`. I vote for keeping this as-is with fewer prompts.

Reply:
Waldek I agree there is no point in having multiple same apps. if someone is does not know why he or she is doing and will just run the setup command multiple times (just for the hell of it to check multiple scenarios) then he or she will end up with a lot of apps. We could just give a small heads up with a warning that you already have an all with same name and do you want to create a new one or overwrite it. 
But to keeps thing simple for now we may leave it as is and discuss it separately in a follow up issue